pkg/controller/storagecluster/storageclasses.go

```diff
@@ -51,7 +51,7 @@ func (r *ReconcileStorageCluster) createStorageClasses(scs []*storagev1.StorageC
 			return err
 		} else {
 			reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.StorageClasses.ReconcileStrategy)
-			if reconcileStrategy == ReconcileStrategyDefault || reconcileStrategy == ReconcileStrategyUnknown {
+			if reconcileStrategy == ReconcileStrategyInit {
```

Comment:
Shouldn't this also return without reconciling if reconcileStrategy is ignore?
Is ReconcileStrategyStandalone only applicable to Nooba?

Ditto for the other changes.

Reply:
The delete<resource> function for each of these resources should check if the reconcileStrategy is ReconcileStrategyInit before attempting to delete them.

Reply:
> Shouldn't this also return without reconciling if reconcileStrategy is ignore?

If we have `ReconcileStrategyIgnore`, then we don't even call this function. This is checked above in `ensureStorageClasses`.  For the other cases, the `Ignore` case is checked further up in the same function. In general: 

* ignore ==> don't call the reconcile logic at all
* init ==> only create if it does not exist
* manage ==> create and update

> Is ReconcileStrategyStandalone only applicable to Nooba?

Correct. @nb-ohad wanted it to behave differently than the other resources.



> The delete function for each of these resources should check if the reconcileStrategy is ReconcileStrategyInit before attempting to delete them.

Hmmm, interesting point. We currently only delete the storage resources in uninstall, not in reconciliation, as it seems. We might need to talk about it, but it is not a new thing.

Reply:
> Shouldn't this also return without reconciling if reconcileStrategy is ignore?
> Is ReconcileStrategyStandalone only applicable to Nooba?

It already does at the beginning of the function.

> The delete function for each of these resources should check if the reconcileStrategy is ReconcileStrategyInit before attempting to delete them.

I would say no, we should be deleting everything during uninstall. There's no reason for the CRs to remain when the StorageCluster is gone.